crates/noirc_evaluator/src/ssa/conditional.rs

```diff
@@ -731,36 +733,25 @@ impl DecisionTree {
     ) -> Option<NodeId> {
         match (condition1, condition2) {
             (None, None) => None,
-            (Some(condition), None) | (None, Some(condition)) => {
-                if condition.is_dummy() {
-                    None
-                } else {
-                    Some(condition)
-                }
+            (Some(cond), other) | (other, Some(cond)) if cond.is_dummy() => {
+                Self::and_conditions(None, other, stack_frame, ctx)
             }
+            (Some(cond), None) | (None, Some(cond)) => Some(cond),
```

Comment:
This is missing the `(Some(cond1), Some(cond2)) if cond1 == cond2` case now. Or was it purposely removed?

Reply:
No I forgot it, good catch!